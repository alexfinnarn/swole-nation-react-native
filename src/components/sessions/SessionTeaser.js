import React from 'react';
import {Text, View} from "react-native";

export default function SessionTeaser({ session, styles }) {
  function getCompletedString() {
    let count = 0;
    session.exercises.forEach((exercise) => {
      count = count + exercise.sets.length;
    });

    const num = (session.completed / count) * 100;
    return `${num.toFixed(0)}%`
  }

  return (
    <View testID="session-teaser-root" style={{flex: 1, flexDirection: 'column', justifyContent: 'space-evenly', ...styles}}>
      <View style={{flex: 1, padding: 10, flexDirection: 'row', justifyContent: 'space-between'}}>
        <Text testID="session-name">{ session.name }</Text>
      </View>
      <View style={{flex: 2, padding: 10, flexDirection: 'row', justifyContent: 'space-between'}}>
        <View style={{flexDirection: 'column', alignItems: 'flex-start'}}>
          <Text>Workout:</Text>
          <Text testID="workout-name">{session.workoutName}</Text>
        </View>
        <View style={{flexDirection: 'column', alignItems: 'center'}}>
          <Text>Completed:</Text>
          <Text testID="completion-percentage">{getCompletedString()}</Text>
        </View>
        <View style={{flexDirection: 'column', alignItems: 'center'}}>
          <Text>Duration:</Text>
          <Text testID="session-duration">{session.duration.toString()}s</Text>
        </View>
      </View>
    </View>
  );
}
