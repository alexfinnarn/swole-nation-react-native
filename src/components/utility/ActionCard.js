import React from 'react';
import {Text, View} from 'react-native';
import {home} from '../Styles';
import ActionButton from './ActionButton';

export default function ActionCard({children, actionComponent}) {
  return (
    <View style={home.sectionContainer}>
      <View style={home.sectionLeft}>{children}</View>
      {actionComponent}
    </View>
  );
}
