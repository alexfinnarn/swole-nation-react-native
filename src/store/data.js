import addFive from '../transformers/addFive'

const data = {
  workouts: {
    DQkECwYLCQQ: {
      name: "Stronglifts A",
      key: "DQkECwYLCQQ",
      description: "Squats, Bench Press, Barbell Row",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Bench Press Warmup",
        "Bench Press",
        "Barbell Row Warmup",
        "Barbell Row"
      ]
    },
    DAwJBQEMAA0: {
      name: "Stronglifts B",
      key: "DAwJBQEMAA0",
      description: "Squats, Overhead Press, Deadlifts",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Overhead Press Warmup",
        "Overhead Press",
        "Deadlifts Warmup",
        "Deadlifts"
      ]
    }
  },
  exercises: {
    BwEBDwoFBgQ: {
      name: "Squats Warmup",
      key: "BwEBDwoFBgQ",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    },
    Bw0ECwAKBws: {
      name: "Squats",
      key: "Bw0ECwAKBws",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        }
      ]
    },
    BwwBDAIJDgw: {
      name: "Bench Press Warmup",
      key: "BwwBDAIJDgw",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 3,
          "weight": 135.0
        }
      ]
    },
    DgcBCA8BCQo: {
      name: "Bench Press",
      key: "DgcBCA8BCQo",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        }
      ]
    },
    DwcACw4PCgI: {
      name: "Shoulder Press Warmup",
      key: "DwcACw4PCgI",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 3,
          "weight": 75.0
        }
      ]
    },
    CAMBAwsFBwQ: {
      name: "Shoulder Press",
      key: "CAMBAwsFBwQ",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        }
      ]
    },
    BwYGAw0NDwo: {
      name: "Barbell Row Warmup",
      key: "BwYGAw0NDwo",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 95.0
        }
      ]
    },
    BwQKAwYKAgA: {
      name: "Barbell Row",
      key: "BwQKAwYKAgA",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 115.0
        },
        {
          "reps": 5,
          "weight": 115.0
        }
      ]
    },
    AQcNAQkBCwo: {
      name: "Deadlifts Warmup",
      key: "AQcNAQkBCwo",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 5,
          "weight": 185.0
        },
        {
          "reps": 5,
          "weight": 205.0
        }
      ]
    },
    DAYCDAQKAw4: {
      name: "Deadlifts",
      key: "DAYCDAQKAw4",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 225.0
        }
      ]
    }
  },
  sessions: {
    BwgFDQ8CCg8: {
      name: "new ...date",
      key: "BwgFDQ8CCg8",
      workoutKeys: [],
      exercises: []
    }
  },
  transformers: {
    none: {
      key: 'none',
      label: "N/A",
      callback: null
    },
    addFive: {
      key: 'addFive',
      label: "Exercises (+5 lbs)",
      callback: addFive
    }
  },
  activeWorkoutKey: "",
  activeExerciseKey: "",
  activeTransformerKey: "none",
};

export default data;
