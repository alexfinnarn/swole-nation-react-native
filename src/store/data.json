{
  "workouts": [
    {
      "name": "Stronglifts A",
      "key": "DQkECwYLCQQ",
      "description": "Squats, Bench Press, Barbell Row",
      "exercises": [
        "Squats Warmup"
      ]
    }
  ],
  "exercises": [
    {
      "name": "Squats Warmup",
      "key": "BwEBDwoFBgQ",
      "image": "../../assets/squats.gif",
      "instructions": "Do with a barbell",
      "sets": [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    }
  ],
  "sessions": [
    {
      "name": "new ...date",
      "key": "BwgFDQ8CCg8",
      "workoutKeys": [],
      "exercises": []
    }
  ],
  "tempExerciseList": [],
  "theThing": "",
  "activeWorkoutIndex": 0,
  "activeSessionIndex": 0,
  "activeExerciseIndex": 0
}
