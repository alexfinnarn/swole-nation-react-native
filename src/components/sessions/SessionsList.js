import React from 'react';
import {FlatList, Text, View} from "react-native";
import {home, styles} from '../Styles';
import ActionButton from "../utility/ActionButton";
import ActionCard from "../utility/ActionCard";
import SessionTeaser from "./SessionTeaser";

export default function SessionsList({ sessions, navigation, handle, thing }) {
  return (
    <View style={styles.container} testID="sessions-list-root">
      <View style={{flex:9}}>
        <FlatList
          extraData={thing}
          data={sessions}
          ListEmptyComponent={(<Text>No sessions to list.</Text>)}
          renderItem={(item) => <SessionsListItem session={item}/>}
        />
      </View>
      {/*<ActionButton text="Add session" action={() => navigation.navigate('Session')} />*/}
    </View>
  );

  function SessionsListItem({session}) {
    return (
      <ActionCard actionComponent={
        <>
          <ActionButton styles={{marginRight: 2}} text="Edit" action={() => {
            handle.setActiveSessionKey(session.item.key);
            navigation.navigate('Session', {sessionId: session.item.key, title: session.item.name})
          }}/>
          <ActionButton text="X" action={() => {
            handle.deleteSession(session.item.key);
            // navigation.navigate('Home');
          }}/>
        </>
      }>
        <SessionTeaser session={session.item}/>
      </ActionCard>
    );
  }
}



