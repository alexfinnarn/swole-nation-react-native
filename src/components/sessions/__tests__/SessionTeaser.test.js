import React from 'react';
import SessionsList from '../SessionsList';
import { render, fireEvent } from 'react-native-testing-library';
import data from '../../../store/data';
import ActionCard from "../../utility/ActionCard";
import shortId from "shortid";
import SessionTeaser from "../SessionTeaser";

let renderer, instance ={};
let session = data.sessions[Object.keys(data.sessions)[0]];
const handle = {
  setActiveSessionKey: jest.fn(),
  deleteSession: jest.fn(),
};
const navigation = {
  navigate: jest.fn()
};

beforeEach(() => {
  renderer = render(<SessionTeaser session={session}/>);
  instance = renderer.getByTestId('session-teaser-root');
}, 0);

describe('<SessionsList />', () => {
  it('Renders correctly', () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('Should render session name', () => {
    expect(renderer.getByTestId('session-name').props.children).toBe('Thu Sep 26 12:19:14 2019');
  });

  it('Should render workout name', () => {
    expect(renderer.getByTestId('workout-name').props.children).toBe('Stronglifts A');
  });

  it('Should render completed percentage', () => {
    expect(renderer.getByTestId('completion-percentage').props.children).toBe('78%');
  });

  it('Should render session duration', () => {
    expect(renderer.getByTestId('session-duration').props.children).toEqual(["15", "s"]);
  });
});
