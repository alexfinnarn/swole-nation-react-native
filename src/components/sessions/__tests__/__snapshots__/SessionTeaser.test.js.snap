// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SessionsList /> Renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-evenly",
    }
  }
  testID="session-teaser-root"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <Text
      testID="session-name"
    >
      Thu Sep 26 12:19:14 2019
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 2,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
        }
      }
    >
      <Text>
        Workout:
      </Text>
      <Text
        testID="workout-name"
      >
        Texas Method A
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
        }
      }
    >
      <Text>
        Completed:
      </Text>
      <Text
        testID="completion-percentage"
      >
        78%
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
        }
      }
    >
      <Text>
        Duration:
      </Text>
      <Text
        testID="session-duration"
      >
        15
        s
      </Text>
    </View>
  </View>
</View>
`;
